From 41d08bbca26fa762aa89f3b9a38ded758a3e65d4 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:55:57 -0400
Subject: [scsi] qla2xxx: Add BPM support for ISP25xx

Message-id: <1408391817-6389-10-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87907
O-Subject: [PATCH RHEL 7.1 e-stor 09/69] qla2xxx: Add BPM support for ISP25xx.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Chad Dupuis <chad.dupuis@qlogic.com>

scsi commit id ca0e68db4ded4da0dc837536f0ca9a16f290c381

Signed-off-by: Chad Dupuis <chad.dupuis@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mbx.c b/drivers/scsi/qla2xxx/qla_mbx.c
index 5a17ab1..d017513 100644
--- a/drivers/scsi/qla2xxx/qla_mbx.c
+++ b/drivers/scsi/qla2xxx/qla_mbx.c
@@ -468,7 +468,7 @@ qla2x00_execute_fw(scsi_qla_host_t *vha, uint32_t risc_addr)
   mcp->mb[1] = MSW(risc_addr);
   mcp->mb[2] = LSW(risc_addr);
   mcp->mb[3] = 0;
-  if (IS_QLA81XX(ha) || IS_QLA83XX(ha)) {
+  if (IS_QLA25XX(ha) || IS_QLA81XX(ha) || IS_QLA83XX(ha)) {
    struct nvram_81xx *nv = ha->nvram;
    mcp->mb[4] = (nv->enhanced_features &
        EXTENDED_BB_CREDITS);
-- 
1.7.1